tests/test_services/test_analytics_service.py

```diff
@@ -0,0 +1,86 @@
+"""Tests for analytics service."""
+import os
+from unittest.mock import PropertyMock, patch
+
+import pytest
+
+from matcha_ml.cli.cli import app
+from matcha_ml.services.global_parameters_service import GlobalParameters
+
+GLOBAL_PARAMETER_SERVICE_FUNCTION_STUB = (
+    "matcha_ml.services.analytics_service.GlobalParameters"
+)
+
+
+@pytest.fixture(autouse=True)
+def mocked_global_parameters_service(matcha_testing_directory):
+    """Mocked global parameters service.
+
+    Args:
+        matcha_testing_directory (str): Temporary directory for testing.
+
+    Yields:
+        GlobalParameters: GlobalParameters object with mocked properties.
+    """
+    with patch(
+        f"{GLOBAL_PARAMETER_SERVICE_FUNCTION_STUB}.default_config_file_path",
+        new_callable=PropertyMock,
+    ) as file_path, patch(
+        f"{GLOBAL_PARAMETER_SERVICE_FUNCTION_STUB}.user_id",
+        new_callable=PropertyMock,
+    ) as user_id:
+        file_path.return_value = str(
+            os.path.join(str(matcha_testing_directory), ".matcha-ml", "config.yaml")
+        )
+        user_id.return_value = "TestUserID"
+
+        yield GlobalParameters()
+
+
+def test_cli_destroy_command_analytics_are_mocked(
```

Comment:
This feels more like a test to see if the `mock` works rather than testing the functionality of our code?

One test that you could add is that the values sent to Segment are what you expect - you'd have to separate out the `analytics.track()` and build a data object to pass to the function to test this. This would have the benefit that we know the analytics being sent are broadly accurate.

Reply:
Updated the test to check the arguments without separating it out